sentry_sdk/worker.py

```diff
@@ -86,6 +86,10 @@ def start(self):
 
     def kill(self):
         # type: () -> None
+        """
+        Kill worker thread. Returns immediately. Not useful for
+        waiting on shutdown for events, use `flush` for that.
+        """
```

Comment:
In theory now `kill` may raise `queue.Full`.

I'll consider people should never call this directly, but instead use `Hub.current.client.close()`, which does:

```python
           self.flush(timeout=timeout, callback=callback)
           self.transport.kill()
```

The first `flush` should empty the queue (may not empty if the timeout is reached), then `transport.kill()` should be able to put the `_TERMINATOR`.

Since there are some "shoulds", how about we handle `queue.Full` explicitly? In that case we can log a debug message. I don't see a clear way out without changing the current termination mechanism.

Thoughts?

Reply:
Yeah I did the same thing as with the other put_nowait now